src/main/java/io/reactivex/internal/operators/parallel/ParallelJoin.java

```diff
@@ -110,15 +110,13 @@ public void cancel() {
         }
 
         void cancelAll() {
-            for (int i = 0; i < subscribers.length; i++) {
-                JoinInnerSubscriber<T> s = subscribers[i];
+            for (JoinInnerSubscriber<T> s : subscribers) {
```

Comment:
:+1:

Reply:
Don't we cop allocation cost for the creation of an iterator?

Reply:
`subscribers` is an array so no.

Reply:
of course, thanks